atomic_reactor/plugins/post_koji_upload.py

```diff
@@ -0,0 +1,643 @@
+from atomic_reactor.util import (get_version_of_tools, get_checksums,
+                                 get_build_json, get_docker_architecture)
+from atomic_reactor.koji_util import create_koji_session
+from osbs.conf import Configuration
+from osbs.api import OSBS
+from osbs.exceptions import OsbsException
+
+# An output file and its metadata
+Output = namedtuple('Output', ['file', 'metadata'])
+
+
+class KojiUploadLogger(object):
+    def __init__(self, logger, notable_percent=10):
+        self.logger = logger
+        self.notable_percent = notable_percent
+        self.last_percent_done = 0
+
+    def callback(self, offset, totalsize, size, t1, t2):  # pylint: disable=W0613
+        if offset == 0:
+            self.logger.debug("upload size: %.1fMiB", totalsize / 1024.0 / 1024)
+
+        if not totalsize or not t1:
+            return
+
+        percent_done = 100 * offset / totalsize
+        if (percent_done >= 99 or
+                percent_done - self.last_percent_done >= self.notable_percent):
+            self.last_percent_done = percent_done
+            self.logger.debug("upload: %d%% done (%.1f MiB/sec)",
+                              percent_done, size / t1 / 1024 / 1024)
+
+
+class KojiUploadPlugin(PostBuildPlugin):
+    """
+    Upload this build to Koji
+
+    Note: only the image archive is uploaded to Koji at this stage.
+    Metadata about this image is created and stored in a ConfigMap in
+    OpenShift, ready for the orchestrator build to collect and use to
+    actually create the Koji Build together with the uploaded image
+    archive(s).
+
+    Authentication is with Kerberos unless the koji_ssl_certs
+    configuration parameter is given, in which case it should be a
+    path at which 'cert', 'ca', and 'serverca' are the certificates
+    for SSL authentication.
+
+    If Kerberos is used for authentication, the default principal will
+    be used (from the kernel keyring) unless both koji_keytab and
+    koji_principal are specified. The koji_keytab parameter is a
+    keytab name like 'type:name', and so can be used to specify a key
+    in a Kubernetes secret by specifying 'FILE:/path/to/key'.
+    """
+
+    key = PLUGIN_KOJI_UPLOAD_PLUGIN_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, kojihub, url,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 blocksize=None, target=None, poll_interval=5):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param kojihub: string, koji hub (xmlrpc)
+        :param url: string, URL for OSv3 instance
+        :param verify_ssl: bool, verify OSv3 SSL certificate?
+        :param use_auth: bool, initiate authentication with OSv3?
+        :param koji_ssl_certs: str, path to 'cert', 'ca', 'serverca'
```

Comment:
Other plugin have `koji_ssl_certs_dir` paramer, only `koji_tag_build` and `koji_promote` have `koji_ssl_cert`.

Lets rename this parameter here, I'll make a PR to update the rest